snapcraft/yaml.py

```diff
@@ -128,7 +128,11 @@ def __init__(self):
                         plugin_name))
 
             if 'after' in properties:
-                after_requests[part_name] = properties.pop('after')
+                afterprop = properties.pop('after')
+                if isinstance(afterprop, str):
```

Comment:
Just to record this here, I think we should just implement this correctly, given that we iterated over the string contents before I don't see how this worked for anyone so it is a plain bug and requires no deprecation message.